raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -76,7 +87,48 @@
     private final int electionBackoffMaxMs;
     private final int fetchTimeoutMs;
     private final int appendLingerMs;
-    private final Map<Integer, InetSocketAddress> voterConnections;
+    private final Map<Integer, AddressSpec> voterConnections;
+
+    public static abstract class AddressSpec {
+       public abstract InetSocketAddress address();
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+
+            final AddressSpec that = (AddressSpec) obj;
+            return that.address().equals(address());
+        }
+    }
+
+    public static class InetAddressSpec extends AddressSpec {
+        private final InetSocketAddress address;
+
+        public InetAddressSpec(InetSocketAddress address) {
+            if (address.equals(UNROUTABLE_ADDRESS)) {
+                throw new IllegalArgumentException("Address not routable");
+            }
+            this.address = address;
+        }
+
+        @Override
+        public InetSocketAddress address() {
+            return address;
+        }
+    }
+
+    public static class UnknownAddressSpec extends AddressSpec {
```

Comment:
A common pattern for classes like this without any state is to create a static instance.
```java
  public static final UnknownAddressSpec INSTANCE = new UnknownAddressSpec();
```

Reply:
Ack. Refactored as stated above.